fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/SizeStatisticsCollector.java

```diff
@@ -351,30 +351,4 @@ public static SizeStatisticsCollector ofIndex(@Nonnull FDBRecordStore store, @No
     public static SizeStatisticsCollector ofSubspace(@Nonnull Subspace subspace) {
         return new SizeStatisticsCollector(subspace);
     }
-
-    /**
-     * Create a statistics collector of all keys used by index within a given {@link SubspaceProvider}'s subspace.
-     * If the implementation of {@link SubspaceProvider#getSubspace() getSubspace()} is blocking for the
-     * given <code>SubspaceProvide</code>, then this method will also be blocking.
-     *
-     * @param subspaceProvider the provider of the subspace to collect statistics on key and value sizes
-     * @return a statistics collector of the given subspace
-     */
-    @Nonnull
-    public static SizeStatisticsCollector ofSubspaceProvider(@Nonnull SubspaceProvider subspaceProvider) {
-        return new SizeStatisticsCollector(subspaceProvider.getSubspace());
```

Comment:
I think we could preserve these two methods if the class remembered a `SubspaceProvider` instead of a `Subspace`. It could also have an `ofKeySpacePath` static initializer, though that's somewhat of a separate task from this PR.

Reply:
These methods are currently unused. Are you suggesting to update these unused methods?

Reply:
The methods themselves I think don't need to be updated; it's the class itself that should have a `SubspaceProvider` instead of a `Subspace`.

This class is intended to be used by clients of the Record Layer rather than by the Record Layer itself. Other than tests of the class itself, I don't believe any of this class's methods are used anywhere in this project.